### PR TITLE
fix panic when reusing a was-empty batch

### LINKED ISSUE #266: Writing an empty batch causes panic
panic message reads: `assertion failed: self.buf_state == BufState::Open`

### PR DESCRIPTION
**Pull request by tabokie**
Close #266

An empty batch went through `finish_populate` but not `drain` (in memtable apply block). This will break internal state machine and panic.

### CODE REVIEW

File: src/engine.rs [hunk 1]
```diff
@@ -136,8 +136,13 @@ where
     /// bytes. If `sync` is true, the write will be followed by a call to
     /// `fdatasync` on the log file.
     pub fn write(&self, log_batch: &mut LogBatch, mut sync: bool) -> Result<usize> {
+        if log_batch.is_empty() {
+            return Ok(0);
+        }
         let start = Instant::now();
         let len = log_batch.finish_populate(self.cfg.batch_compression_threshold.0 as usize)?;
+        debug_assert!(!log_batch.is_empty());
```
Comment (LykxSassinator) [inline src/engine.rs:144]: Do it before `finish_populate` or remove it. The first validation `log_batch.is_empty()` has guaranteed that `log_batch.is_empty() == false`.

**Review by LykxSassinator — COMMENTED** [review]
Rest LGTM

File: src/engine.rs (hunk 1 above)
Comment (tabokie) [inline src/engine.rs:144]:
```suggestion
```

**Review by Connor1996 — APPROVED** [review]
LGTM

**Pull request merged**